pubnub/lib/src/crypto/crypto.dart

```diff
@@ -1,132 +1,106 @@
+        cryptors: <ICryptor>[
+          LegacyCryptor(cipherKey,
+              cryptoConfiguration: defaultCryptoConfiguration)
+        ]);
   }
 
-  /// Decrypts [input] with [key] based on [configuration].
-  ///
-  /// If [configuration] is `null`, then [CryptoModule.defaultConfiguration] is used.
   @override
-  dynamic decrypt(CipherKey key, String input,
-      {CryptoConfiguration? configuration}) {
-    var config = configuration ?? defaultConfiguration;
-
-    var encrypter = crypto.Encrypter(
-        crypto.AES(_getKey(key, config), mode: config.encryptionMode.value()));
-    try {
-      if (config.useRandomInitializationVector) {
-        var data = base64.decode(input);
-        return utf8.decode(encrypter.decryptBytes(
-            crypto.Encrypted(Uint8List.fromList(data.sublist(16))),
-            iv: crypto.IV.fromBase64(base64.encode(data.sublist(0, 16)))));
-      } else {
-        var iv = crypto.IV.fromUtf8('0123456789012345');
-        return encrypter.decrypt64(input, iv: iv);
-      }
-    } catch (e) {
-      throw CryptoException('Error while decrypting message:\n$e');
-    }
+  List<int> encrypt(List<int> data) {
+    var encrypted = defaultCryptor.encrypt(data);
+    if (encrypted.metadata == null) return encrypted.data;
+
+    var header =
+        CryptorHeader.from(defaultCryptor.identifier, encrypted.metadata!);
+    var headerData = List<int>.filled(header!.length, 0);
+    var pos = 0;
+    headerData.setAll(pos, header.data);
+    pos = header.length - encrypted.metadata!.length;
+    headerData.setAll(pos, encrypted.metadata!);
+    return [...headerData, ...encrypted.data];
   }
 
-  /// Encrypts [input] with [key] based on [configuration].
-  ///
-  /// If [configuration] is `null`, then [CryptoModule.defaultConfiguration] is used.
   @override
-  String encrypt(CipherKey key, dynamic input,
-      {CryptoConfiguration? configuration}) {
-    var config = configuration ?? defaultConfiguration;
+  List<int> decrypt(List<int> data) {
+    var header = CryptorHeader.tryParse(data);
+    var cryptor = _getCryptor(header);
+    var headerLength = header != null ? header.length : 0;
+    var metadata = headerLength > 0
+        ? data.sublist((headerLength - header!.metadataLength), headerLength)
+        : null;
+    return cryptor!
+        .decrypt(EncryptedData.from(data.sublist(headerLength), metadata));
+  }
 
-    var encrypter = crypto.Encrypter(
-        crypto.AES(_getKey(key, config), mode: config.encryptionMode.value()));
+  ICryptor? _getCryptor(CryptorHeaderV1? header) {
     try {
-      if (config.useRandomInitializationVector) {
-        var iv = crypto.IV.fromSecureRandom(16);
-
-        var encrypted = encrypter.encrypt(input, iv: iv).bytes;
-        return base64.encode([...iv.bytes, ...encrypted]);
-      } else {
-        var iv = crypto.IV.fromUtf8('0123456789012345');
-        return encrypter.encrypt(input, iv: iv).base64;
-      }
+      var allCryptors = _getAllCryptor();
```

Comment:
Perhaps you could simplify instantiating local `cryptor` variable in this way: 

```
var cryptor = allCryptors.firstWhere(
  (element) => element.identifier == (header?.identifier ?? ''),
);
```

Reply:
good catch